src/routes/jira/server/jira-server-url-post.ts

```diff
@@ -0,0 +1,32 @@
+import { Request, Response } from "express";
+import axios from "axios";
+import { GitHubServerApp } from "models/github-server-app";
+
+export const JiraServerUrlPost = async (
+	req: Request,
+	res: Response
+): Promise<void> => {
+	// todo - validate url
+	const { gheServerURL, installationId } = req.body;
+	req.log.info(`Verifying provided GHE server url: ${gheServerURL}`, installationId);
+
+	try {
+		const gitHubServerApps = await GitHubServerApp.getAllForGitHubBaseUrl(gheServerURL, installationId);
+
+		if (gitHubServerApps?.length) {
+			req.log.info(`GitHub apps found for url: ${gheServerURL}. Redirecting to Jira list apps page.`);
```

Comment:
Changed the order of checks here from validating the url and then checking the GHSAs table to:

- validating that `gheServerURL` is a valid url
- checking if there are apps in the table for the provided url and installationId. If apps exist for the url then we have already succeeded in successfully making a request to that GHE server instance
- if there are no apps in the table and we're coming from an empty state, we also make a request to the url to see if we can get a 200 response (hole in the firewall has been opened).